Mathlib/RingTheory/Ideal/Maps.lean

```diff
@@ -910,3 +910,69 @@ lemma coe_ideal_map (I : Ideal A) :
     Ideal.map f I = Ideal.map (f : A →+* B) I := rfl
 
 end AlgHom
+
+noncomputable section IsPrincipal
+
+namespace Ideal
+
+open nonZeroDivisors Submodule
+
+variable (R : Type*) [CommRing R] [IsDomain R]
+
+/-- The equivalence between `Associates R` and the principal ideals of `R` defined by sending the
```

Comment:
Why do you prefer this theorem to be here? It is not about a ring homomorphism, I find it more appropriate for `Basic`.

Reply:
Well, the new lemmas have to be in this file since otherwise I need to modify the imports in `Ideal.Basic`. Also they are closely related to `Ideal.associatesEquivIsPrincipal`, so I have decided to move it to this file. Finally, all these results are about `maps` on ideals so the file `Ideal.Maps` seemed more adequate :wink: